modules/apps/portal-security/portal-security-service-access-policy-service/src/main/java/com/liferay/portal/security/service/access/policy/internal/upgrade/v3_0_1/SAPEntryUpgradeProcess.java

```diff
@@ -95,6 +103,10 @@ protected void doUpgrade() throws Exception {
 								companyId)));
 
 					preparedStatement2.execute();
+
+					ResourceLocalServiceUtil.addResources(
+						companyId, 0, guestUserId, SAPEntry.class.getName(),
+						sapEntryId, false, false, false);
```

Comment:
Hi @ChrisKian . Curious why you decided to call `ResourcLocalServiceUtil#addResources` here and in `SAPEntryLocalServiceImpl#checkSystemSAPEntries` you called `ResourcePermissionLocalService#setResourcePermissions` for the same purpose?

Reply:
I just used the code from Pedro's script, as it seems to work fine.

I will adjust this code to follow the checkSystemSAPEntries logic more closely.  Thanks for catching this